swift_browser_ui/common/vault_client.py

```diff
@@ -0,0 +1,295 @@
+"""Vault c4ghtransit client"""
```

Comment:
i am wondering if it would have been a good idea to use an already available client instead of implementing our own e.g. `hvac` library seems to do most of the functions below ?

maybe a custom adaptor can be extended: https://hvac.readthedocs.io/en/stable/advanced_usage.html#custom-requests-http-adapter ?

Reply:
welll hvac is not async ... but i am curios why it was not chosen

Reply:
One of the main reasons was that it's not async.

But then the API we're accessing is custom-made. And even with `hvac` We'd need very similar code to what we have now. Perhaps the auth would have been simplified, but it doesn't automatically renew tokens, so that logic still had to be implemented. We'd still need the custom error handling, unless I missed something from their codebase.

I think the code is more straightforward this way, too. Nobody has to learn yet another library to make changes to the code, and there's no hidden magic going on.

But if you think that would be an added benefit to using `hvac`, we could re-evaluate, of course.

Reply:
Or keep the library updated and the code compatible with it. It just felt like the benefits didn't add up enough in this case.

Reply:
for token renew is: https://github.com/hvac/hvac/blob/v1.0.2/hvac/api/auth_methods/token.py#L348 - maybe that is what you are looking for.

> But if you think that would be an added benefit to using hvac, we could re-evaluate, of course.

no, was just curious of the reasons, I understand them. I am ok with the current solution.